### PR TITLE
Skip reference warnings to

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: test/fixtures/warnings.ex [hunk 1]
```diff
@@ -20,3 +20,21 @@ defmodule Warnings do
...
+  @moduledoc """
+  Refer to:
+  - [this file](unknown.md)
+  - [this module](`UknownModule`)
+  - [this function](`CompiledWithDocs.function/0`)
+  - [this callback](`c:CompiledWithDocs.callback/1`)
+  - [this type](`t:CompiledWithDocs.type/2`)
+
+  Refer to with Elixir namespace:
+  - [this module](`Elixir.UknownModule`)
```
Comment (maartenvanvliet) [inline test/fixtures/warnings.ex:34]:
```suggestion
  - [this module](`Elixir.UnknownModule`)
```

File: test/fixtures/warnings.ex [hunk 2]
```diff
@@ -20,3 +20,21 @@ defmodule Warnings do
   """
   def foo(), do: :ok
 end
+
+defmodule Warnings.Submodule do
+  @moduledoc """
+  Refer to:
+  - [this file](unknown.md)
+  - [this module](`UknownModule`)
```
Comment (maartenvanvliet) [inline test/fixtures/warnings.ex:28]:
```suggestion
  - [this module](`UnknownModule`)
```

Comment (eksperimental) [reply test/fixtures/warnings.ex]: Fixed. Thank you!

**Comment by eksperimental** [comment]
There are two features that can be discussed, but are not intended to be covered by this PR:

1. Whether we want to support Mix tasks. ie. "mix foo", instead of using the Mix.Tasks.TASKNAME module.

2. Whether we want to skip references to a a module and everything inside (functions, types, etc). As of now, listing the Module would ignore the warning for that module in particular, but nothing else.

**Comment by josevalim** [comment]
Do you have any example of when we would use this feature?

**Comment by eksperimental** [comment]
Closes #1192
Addresses #1151 (comment)

**Comment by eksperimental** [comment]
Basically linking to hidden reference (i.e. with doc false), such as in https://github.com/elixir-lang/ex_doc/pull/1151#issuecomment-646730079 

where calling `ExUnit.Case.__using__/1`, which is a macro with `@doc false` and it will emit a warning when referencing it.

**Comment by josevalim** [comment]
@eksperimental oh, I see. I wonder if we can tackle this in some other way. Having to change a completely unrelated config file for a mistake happening in the documentation elsewhere is not very clear. Maybe we can use triple backticks to denote code sections with no syntax highlighting. We would need to change markdown first. For now, I will fix this in Elixir by referring to the function indirectly. Thank you!

**Pull request closed without merging**